### PR TITLE
Fix dev setup scripts + related docs for dev portal, remove autogenerated assets

### PR DESCRIPTION
**Pull request by amazon-meaisiah**
The script and docs fixes were based on the steps I took to resolve my issues getting set up, and I wanted to ensure that future potential contributors could also benefit from my work.

~~Please let me know if the autogenerated assets need to remain in Git, and I'll fix the patch accordingly.~~ Reverted that and just listed them as binary so they won't show in diffs.

*Issue #, if available:* #343 

*Description of changes:* The title should explain it well enough.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

**Edit:** add issue

### CODE REVIEW

File: dev-portal/scripts/deploy-stack.js [hunk 1]
```diff
@@ -1,48 +1,72 @@
...
-  })
+const awsSamCliProfile = getOptional('awsSamCliProfile')
+
+async function main() {
+  execute('sam', [
+    'package',
+    '--template-file', samTemplate,
+    '--output-template-file', packageConfig,
+    '--s3-bucket', buildAssetsBucket,
+    ...awsSamCliProfile ? ['--profile', awsSamCliProfile] : []
```
Comment (MynockSpit) [inline dev-portal/scripts/deploy-stack.js]: Please wrap this w/ parens to make this more clear.

File: dev-portal/scripts/deploy-stack.js [hunk 2]
```diff
@@ -1,48 +1,72 @@
...
+    `ArtifactsS3BucketName=${apiAssetsBucket}`,
+    `DevPortalCustomersTableName=${customersTableName}`,
+    `CognitoDomainNameOrPrefix=${cognitoDomainName}`,
+    '--s3-bucket', buildAssetsBucket,
+    ...awsSamCliProfile ? ['--profile', awsSamCliProfile] : []
+  ])
+  await writeConfig()
+  console.log()
+  console.log(green('Process Complete! Run `npm run start` to launch run the dev portal locally.'))
+  console.log()
```
Comment (MynockSpit) [inline dev-portal/scripts/deploy-stack.js]: For spacing? Why not `console.log('\n', blah, '\n')`?

File: .gitignore [hunk 3]
```diff
@@ -17,6 +17,9 @@ dev-portal/node_modules/swagger-ui/*
 
 # built artifacts
 /dev-portal/build
+/lambdas/static-asset-uploader/build/static/js/*.chunk.js
+/lambdas/static-asset-uploader/build/static/js/*.chunk.js.map
+/lambdas/static-asset-uploader/build/precache-manifest.*.js
```
Comment (MynockSpit) [inline .gitignore]: These need to be committed.

File: dev-portal/scripts/deploy-stack.js (hunk 2 above)
Comment (amazon-meaisiah) [reply dev-portal/scripts/deploy-stack.js]: That and to make it align more roughly with what's actually printed to console.

**Pull request merged**